tests/gds_sanity.c

```diff
@@ -190,6 +190,8 @@ int main(int argc, char *argv[])
 
                         uint32_t src_data[CHUNK_SIZE] = {1, 2, 3};
 
+                        mem_type = GDS_MEMORY_GPU;
```

Comment:
mem_type is correctly set above.
There is a known bug for host memory when it has been registered already.
We should fix that bug. Can you file an issue for it (if not present yet)?

Reply:
This situation is different from https://github.com/gpudirect/libgdsync/issues/25 

Running the example without the -g option, memory type is GDS_MEMORY_HOST.
Within gds_alloc_mapped_memory(), after a posix_memalign() (not a cudaMallocHost), gds_map_mem() registers the h_ptr with cuMemHostRegister() and return a d_ptr with cuMemHostGetDevicePointer().
Only the h_ptr is included in the rangeset rset (file memmgr.cpp, line 213) used in gds_map_mem() to find already registered addresses.

Later in the code, when setting descriptors like:
```
descs[k].tag = GDS_TAG_WRITE_VALUE32;
descs[k].write32.ptr = d_dbg;
descs[k].write32.value = 0xA000|i;
descs[k].write32.flags = mem_type;
```
the example is using the d_ptr with mem_type=GDS_MEMORY_HOST. Thus, within gds_stream_post_descriptors() -> gds_fill_poke() -> gds_map_mem() , libgdsync tries to register again the d_ptr (host memory) because previously gds_register_mem_internal() registered only h_ptr rather than d_ptr.

For this reason there is a registration error if mem_type=GDS_MEMORY_HOST.
Let me know if you have the same issue running the example without the -g option; unfortunately, now I can't run and verify this behavior.